walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -257,18 +291,193 @@ public void notEnoughRoom() {
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryAttendeesNoGaps() {
+
+    // Two optional attendees with the no gaps in their schedules
+    // Should return no times
+    //
+    // Events  : |--A--||------A-------| 
+    // Events  : |---------B-----------|
+    // Day     : |---------------------|
+    // Options : 
+
+    Collection<Event> events =
+        Arrays.asList(
+            new Event(
+                "Event 1",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, false),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 2",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+                Collections.singletonList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Collections.emptyList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+
+    Collection<TimeRange> expected = Collections.emptyList();
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryAttendeesGaps() {
+
+    // Two optional attendees with the following gaps in their schedules
+    // Should return time slots that both attendees can attend
+    //
+    // Events  :      |--A--| 
+    // Events  :            |-B-|  |-B-|
+    // Day     : |---------------------|
+    // Options : |----|     
+    // Options :                |--|
+
+      
+    Collection<Event> events =
+        Arrays.asList(
+            new Event(
+                "Event 1",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0930AM, false),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 2",
+                TimeRange.fromStartDuration(TIME_1000AM, 30),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 3",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(16, 0), 30),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 4",
+                TimeRange.fromStartEnd(
+                    TimeRange.getTimeInMinutes(17, 0), TimeRange.END_OF_DAY, true),
+                Collections.singletonList(PERSON_A)),
+            new Event(
+                "Event 1.5",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(9, 0), 30),
+                Collections.singletonList(PERSON_B)),
+            new Event(
+                "Event 5",
+                TimeRange.fromStartDuration(TimeRange.getTimeInMinutes(14, 0), 120),
+                Collections.singletonList(PERSON_B)));
```

Comment:
Do the events in this list align with the graphic in the comment at the top of the method?

A is busy except for one span so I would expect A to have two events total, not four.

Reply:
I was trying to make the visual as simple as possible as per tiancong's request to express what the test would be doing but I can change it if you'd like

Reply:
Thanks. The test and its documentation should match.